signalflow/computation.go

```diff
@@ -206,11 +214,41 @@ func (c *Computation) bufferDataMessages() {
 	}
 }
 
+// Buffer up expiration messages indefinitely until another goroutine reads
+// them off of c.expirationCh, which is an unbuffered channel.
+func (c *Computation) bufferExpirationMessages() {
+	buffer := make([]*messages.ExpiredTSIDMessage, 0)
+	var nextMessage *messages.ExpiredTSIDMessage
+	for {
+		if len(buffer) > 0 {
+			if nextMessage == nil {
+				nextMessage, buffer = buffer[0], buffer[1:]
+			}
+			select {
+			case <-c.ctx.Done():
+				return
+			case c.expirationCh <- nextMessage:
```

Comment:
If `expirationCh` is unbuffered won't this block until something reads from it? (which in turn blocks `expirationChBuffer`, which blocks the overall socket if a `ExpiredTSIDMessage` comes in?)

Reply:
The switch statement blocks as a whole until one of the three cases are ready (the ordering of cases does not matter, in fact I think it randomizes the selection process a bit if multiple cases are ready when you get into the select).  The next case of reading from expirationChBuffer will continue if a message comes in on it, and this case will only fire if c.expirationCh can be written to.

Reply:
Oh, I forgot that `select` helps with non-blocking. Ok, that seems fine then so long as it's correct on both sides.